core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -108,30 +108,34 @@ export class AdminBackendApiService {
-    return this.http.get<AdminPageDataBackendDict>(
-      AdminPageConstants.ADMIN_HANDLER_URL).toPromise().then(response => {
-      return {
-        demoExplorations: response.demo_explorations,
-        demoCollections: response.demo_collections,
-        demoExplorationIds: response.demo_exploration_ids,
-        oneOffJobStatusSummaries: response.one_off_job_status_summaries.map(
-          this.jobStatusSummaryObjectFactory.createFromBackendDict),
-        humanReadableCurrentTime: response.human_readable_current_time,
-        auditJobStatusSummaries: response.audit_job_status_summaries.map(
-          this.jobStatusSummaryObjectFactory.createFromBackendDict),
-        updatableRoles: response.updatable_roles,
-        roleGraphData: response.role_graph_data,
-        configProperties: response.config_properties,
-        viewableRoles: response.viewable_roles,
-        unfinishedJobData: response.unfinished_job_data.map(
-          this.jobDataObjectFactory.createFromBackendDict),
-        recentJobData: response.recent_job_data.map(
-          this.jobDataObjectFactory.createFromBackendDict),
-        continuousComputationsData: response.continuous_computations_data.map(
-          this.computationDataObjectFactory.createFromBackendDict),
-        topicSummaries: response.topic_summaries.map(
-          this.topicSummaryObjectFactory.createFromBackendDict)
-      };
+    return new Promise((resolve, reject) => {
+      this.http.get<AdminPageDataBackendDict>(
+        AdminPageConstants.ADMIN_HANDLER_URL).toPromise().then(response => {
+        resolve({
+          demoExplorations: response.demo_explorations,
+          demoCollections: response.demo_collections,
+          demoExplorationIds: response.demo_exploration_ids,
+          oneOffJobStatusSummaries: response.one_off_job_status_summaries.map(
+            this.jobStatusSummaryObjectFactory.createFromBackendDict),
+          humanReadableCurrentTime: response.human_readable_current_time,
+          auditJobStatusSummaries: response.audit_job_status_summaries.map(
+            this.jobStatusSummaryObjectFactory.createFromBackendDict),
+          updatableRoles: response.updatable_roles,
+          roleGraphData: response.role_graph_data,
+          configProperties: response.config_properties,
+          viewableRoles: response.viewable_roles,
+          unfinishedJobData: response.unfinished_job_data.map(
+            this.jobDataObjectFactory.createFromBackendDict),
+          recentJobData: response.recent_job_data.map(
+            this.jobDataObjectFactory.createFromBackendDict),
+          continuousComputationsData: response.continuous_computations_data.map(
+            this.computationDataObjectFactory.createFromBackendDict),
+          topicSummaries: response.topic_summaries.map(
+            this.topicSummaryObjectFactory.createFromBackendDict)
+        });
+      }, errorResponse => {
+        reject(errorResponse.error.error);
+      });
```

Comment:
I've refactored 
```typescript
return this.http.get(...).then(response => {
  ...
  return something;
}, error {
  throw new Error(error.error.error);
});
```

to 

```typescript
return new Promise((resolve, reject) => {
  this.http.get(...).then(response => {
    ...
    resolve(something);
  }, error {
    reject(error.error.error);
  });
});
```

Because the first one when rejecting the promise gives an instance of `Error` object. but the second one gives a string. In almost every other places in the codebase we return a string in case of a rejection. So, I did this to bring consistency. Also in many places we use `AlertsService` to throw errors from backend. This makes it easy to do that.